pyproject.toml

```diff
@@ -68,7 +68,7 @@ exclude = [
 line-length = 100
 target-version = "py311"
 select = ["F", "E", "W", "I", "UP", "S", "B", "T20", "RUF"]
-ignore = ["E501", "B904", "B007", "S308", "S113", "S314", "S603", "S607", "RUF012"]
+ignore = ["E501", "B904", "B007", "S308", "S113", "S314", "S603", "S607", "RUF012", "RUF015"]
```

Comment:
What's your reason for ignoring this rule? Because its slightly less readable to do `next(iter(my_list))`?
https://docs.astral.sh/ruff/rules/unnecessary-iterable-allocation-for-first-element/

Reply:
Yes, exactly! If I was a normal python developer I'd be very confused by `next(iter(...))` - I'd be fine we made some helper method get_first(...) or something, but I figured in the cases where we slice the whole list, we're not doing anything too computationally expensive.